### PR TITLE
Added merkle tree

### PR DESCRIPTION
**Pull request by kevintaing86**
please excuse the branch name. I added merkle tree helper function

### CODE REVIEW

File: block/helpers.py [hunk 1]
```diff
@@ -1,2 +1,28 @@
...
+
+def hashPairs(id1, id2):
+
+    # little and big endian stuff
+    x = id1[::-1]
+    y = id2[::-1]
+
+    return doubleHash(x + y)[::-1]
+
+def findMerkleRoot(myList):
```
Comment (dlowrey) [inline block/helpers.py]: Missing docstring, see https://google.github.io/styleguide/pyguide.html#Comments

File: block/helpers.py [hunk 2]
```diff
@@ -1,2 +1,28 @@
...
+    # base case: if only 1 item, return that item
+    if len(myList) == 1:
+        return myList[0]
+
+    newList = []
+    for i in range(0, len(myList) - 1, 2):
```
Comment (dlowrey) [inline block/helpers.py]: Unusual for loop, needs more comments explaining what its for

File: block/helpers.py [hunk 3]
```diff
@@ -1,2 +1,28 @@
 # A set of helper functions for calculations etc
+import hashlib
 
+def doubleHash(myString):
+    return hashlib.sha256(hashlib.sha256(myString.encode('utf-8')).hexdigest().encode('utf-8')).hexdigest()
+
+def hashPairs(id1, id2):
+
```
Comment (dlowrey) [inline block/helpers.py]: missing docstring see https://google.github.io/styleguide/pyguide.html#Comments

File: block/helpers.py [hunk 4]
```diff
@@ -1,2 +1,28 @@
 # A set of helper functions for calculations etc
+import hashlib
 
+def doubleHash(myString):
+    return hashlib.sha256(hashlib.sha256(myString.encode('utf-8')).hexdigest().encode('utf-8')).hexdigest()
+
+def hashPairs(id1, id2):
+
+    # little and big endian stuff
```
Comment (dlowrey) [inline block/helpers.py]: Better comments

**Comment by dlowrey** [comment]
Maybe it's because I don't fully understand the purpose of the `findMerkleRoot` function, but since you just import and call it from the block class I think it's implementation should also be in the block class, instead of in a helper file.

**Comment by kevintaing86** [comment]
Block class calls merkle tree functions, but merkle tree stuff is not in the same scope as block. Think of merkle tree code as a library that block class is using

**Comment by kevintaing86** [comment]
@dlowrey

**Comment by dlowrey** [comment]
What other methods is a Merkle tree going to have? If it is a data structure we are going to be using I would rather it be in it's own class.

File: block/block.py [hunk 5]
```diff
@@ -26,14 +27,14 @@ def __init__(self, previous_hash=None, transactions=None):
         # define block header attributes
         self.version = version.encode('utf8')  # 4 bytes
         self.previous_hash = previous_hash.encode('utf8')  # 32 bytes
+        self.transactions = transactions  # NOTE: may need to change
```
Comment (leesander1) [inline block/block.py]: Why in header? should be in block body? reason for change?

File: block/helpers.py [hunk 6]
```diff
@@ -1,2 +1,76 @@
 # A set of helper functions for calculations etc
```
Comment (leesander1) [inline block/helpers.py:1]: looks good.

File: block/block.py [hunk 7]
```diff
@@ -26,14 +27,14 @@ def __init__(self, previous_hash=None, transactions=None):
         # define block header attributes
         self.version = version.encode('utf8')  # 4 bytes
         self.previous_hash = previous_hash.encode('utf8')  # 32 bytes
-        self.merkle_root = self.merkle_root()  # 32 bytes
+        self.merkle_root = self.merkle_root(transactions)  # 32 bytes
```
Comment (leesander1) [inline block/block.py:30]: why do we need to pass transactions to the merkle_root function if the transactions themselves are stored in the block object and accessible under self?

Comment (leesander1) [reply block/block.py:30]: Oh I see... Ok coolio

**Pull request merged**